Improve the same user check and its documentation (#69844)
elastic/elasticsearch: docs/reference/search/async-search.asciidoc

The get async search API retrieves the results of a previously submitted
async search request given its id. If the {es} {security-features} are enabled,
the access to the results of a specific async search is restricted to

> <<same-user-check,the user that submitted it in the first place>>.

Reviewer: ```suggestion
<<same-user-check,the user or API Key that submitted it>>
```

> <<can-access-resources-check,the user or API Key that submitted it>>.

Reviewer 2: ```suggestion
<<can-access-resources-check,the user or API key that submitted it>>.
```
